### PR TITLE
fix: Compatible with glib 2.5.58 < 2.60

### PR DESCRIPTION
**Pull request by xingwozhonghua126**
issue: https://github.com/fcitx/fcitx5/issues/459

### CODE REVIEW

File: gtk2/fcitximcontext.cpp [hunk 1]
```diff
@@ -430,7 +430,17 @@ static void fcitx_im_context_finalize(GObject *obj) {
...
     g_clear_pointer(&context->surrounding_text, g_free);
     g_clear_pointer(&context->attrlist, pango_attr_list_unref);
-    g_queue_clear_full(&context->gdk_events, (GDestroyNotify)gdk_event_free);
+    /* https://github.com/GNOME/glib/blob/main/glib/gqueue.c#L164
+     * Compatible with glib 2.5.58 < 2.60
+     * g_queue_clear_full(&context->gdk_events,
+     * (GDestroyNotify)gdk_event_free);*/
+    g_return_if_fail(&context->gdk_events != NULL);
+
+    if ((GDestroyNotify)gdk_event_free != NULL) {
```
Comment (wengxt) [inline gtk2/fcitximcontext.cpp]: This line is useless

File: gtk3/fcitximcontext.cpp [hunk 2]
```diff
@@ -495,7 +495,17 @@ static void fcitx_im_context_finalize(GObject *obj) {
...
     g_clear_pointer(&context->surrounding_text, g_free);
     g_clear_pointer(&context->attrlist, pango_attr_list_unref);
-    g_queue_clear_full(&context->gdk_events, (GDestroyNotify)gdk_event_free);
+    /* https://github.com/GNOME/glib/blob/main/glib/gqueue.c#L164
+     * Compatible with glib 2.5.58 < 2.60
+     * g_queue_clear_full(&context->gdk_events,
+     * (GDestroyNotify)gdk_event_free);*/
+    g_return_if_fail(&context->gdk_events != NULL);
+
+    if ((GDestroyNotify)gdk_event_free != NULL) {
```
Comment (wengxt) [inline gtk3/fcitximcontext.cpp]: This line is useless

File: gtk3/fcitximcontext.cpp [hunk 3]
```diff
@@ -495,7 +495,17 @@ static void fcitx_im_context_finalize(GObject *obj) {
     g_clear_pointer(&context->preedit_string, g_free);
     g_clear_pointer(&context->surrounding_text, g_free);
     g_clear_pointer(&context->attrlist, pango_attr_list_unref);
-    g_queue_clear_full(&context->gdk_events, (GDestroyNotify)gdk_event_free);
+    /* https://github.com/GNOME/glib/blob/main/glib/gqueue.c#L164
+     * Compatible with glib 2.5.58 < 2.60
+     * g_queue_clear_full(&context->gdk_events,
+     * (GDestroyNotify)gdk_event_free);*/
+    g_return_if_fail(&context->gdk_events != NULL);
```
Comment (wengxt) [inline gtk3/fcitximcontext.cpp]: This is also useless.

File: gtk2/fcitximcontext.cpp [hunk 4]
```diff
@@ -430,7 +430,17 @@ static void fcitx_im_context_finalize(GObject *obj) {
     g_clear_pointer(&context->preedit_string, g_free);
     g_clear_pointer(&context->surrounding_text, g_free);
     g_clear_pointer(&context->attrlist, pango_attr_list_unref);
-    g_queue_clear_full(&context->gdk_events, (GDestroyNotify)gdk_event_free);
+    /* https://github.com/GNOME/glib/blob/main/glib/gqueue.c#L164
+     * Compatible with glib 2.5.58 < 2.60
+     * g_queue_clear_full(&context->gdk_events,
+     * (GDestroyNotify)gdk_event_free);*/
+    g_return_if_fail(&context->gdk_events != NULL);
```
Comment (wengxt) [inline gtk2/fcitximcontext.cpp]: just think about it, how can &XX == null?

**Review by wengxt — CHANGES_REQUESTED** [review]

File: gtk2/fcitximcontext.cpp (hunk 4 above)
Comment (xingwozhonghua126) [reply gtk2/fcitximcontext.cpp]: When the gtk event is empty, it cannot enter here, so there is no need to judge, so these codes are useless

**Pull request merged**